janusgraph-core/src/main/java/org/janusgraph/graphdb/database/util/StaleIndexRecordUtil.java

```diff
@@ -0,0 +1,313 @@
+ * IndexRecordEntry[] fullIndexRecord = new IndexRecordEntry[]{new IndexRecordEntry(relationId, value, propertyKey)};
+ * InternalVertex internalVertex1 = (InternalVertex) edge.outVertex();
+ * InternalVertex internalVertex2 = (InternalVertex) edge.inVertex();
+ * JanusGraphElement elementToBeRemoved = new StandardEdge(relationId, edgeLabel, internalVertex1, internalVertex2, ElementLifeCycle.New);
+ * JanusGraphIndex indexToBeUpdated = managementSystem.getGraphIndex(compositeIndexName);
+ * JanusGraphSchemaVertex indexSchemaVertex = managementSystem.getSchemaVertex(indexToBeUpdated);
+ * CompositeIndexType compositeIndexTypeToBeUpdated = (CompositeIndexType) indexSchemaVertex.asIndexType();
+ * Serializer serializer = graph.getDataSerializer();
+ * boolean hashKeys = graph.getIndexSerializer().isHashKeys();
+ * HashingUtil.HashLength hashLength = graph.getIndexSerializer().getHashLength();
+ * IndexUpdate<StaticBuffer, Entry> update = IndexRecordUtil.getCompositeIndexUpdate(
+ *     compositeIndexTypeToBeUpdated,
+ *     IndexMutationType.DELETE,
+ *     fullIndexRecord,
+ *     elementToBeRemoved,
+ *     serializer,
+ *     hashKeys,
+ *     hashLength
+ * );
+ * BackendTransaction backendTransaction = tx.getTxHandle();
+ * backendTransaction.mutateIndex(update.getKey(), Collections.emptyList(), Collections.singletonList(update.getEntry()));
+ * transaction.commit();
+ * tx.commit();
+ * mgmt.rollback();
+ * </pre>
+ *
+ */
+public class StaleIndexRecordUtil {
+
+    /**
+     * Force removes vertex record from a graph index. Notice, only Composite indices are supported by this method at the moment.
+     *
+     * @param vertexId the vertex which should be removed from a specified index
+     * @param indexRecordPropertyValues all property values of the index record
+     * @param graph JanusGraph instance to be used to open graph management and new backend transaction for index removal.
+     * @param graphIndexName name of the index for which to remove a record
+     * @throws BackendException is thrown in case backend transaction cannot be mutated for any reason.
+     */
+    public static void forceRemoveVertexFromGraphIndex(long vertexId,
+                                                       Map<String, Object> indexRecordPropertyValues,
+                                                       JanusGraph graph,
+                                                       String graphIndexName) throws BackendException {
+
+        graph.tx().rollback();
+        ManagementSystem managementSystem = (ManagementSystem) graph.openManagement();
+
+        try{
+            JanusGraphIndex index = managementSystem.getGraphIndex(graphIndexName);
+
+            PropertyKey[] propertyKeys = index.getFieldKeys();
+
+            IndexRecordEntry[] indexRecord = toIndexRecord(propertyKeys, indexRecordPropertyValues);
+
+            StandardJanusGraphTx tx = (StandardJanusGraphTx) graph.newTransaction();
+
+            try{
```

Comment:
In case `mutate` fails we will have a zombie transaction which won't be closed by anyone (unless someone scans, finds it a closes). 
I don't think merging them would be nice because managementSystem has it's own separate transaction. If one fails it doesn't mean we need to skip closing `managementSystem`

Reply:
What I meant is to move `StandardJanusGraphTx tx = (StandardJanusGraphTx) graph.newTransaction();` out of the try-finally block and then we don't need the inner try-finally block. The `tx.rollback()` can be put in the same `finally` block as `managementSystem.rollback()`. Anyways, the code logic remains the same so it does not matter and I am fine leaving this code as it is.

Reply:
I don't know if it is a read case or not but in such case if one `.rollback()` fails for whatever reason another one will not be executed. As I said, I don't know if that's a real case because I didn't see `rollback` throwing an exception for an opened transaction. Moreover, if `StandardJanusGraphTx tx = (StandardJanusGraphTx) graph.newTransaction();` fails for whatever reason then we never reach `try-finally` block and thus we never execute `managementSystem.rollback()` unless I missed anything. 
I will leave this part as it is just in case. I don't mind moving it out of `try-finally` block in another PR if someone thinks it's safe.

Reply:
Got it. My suggestion was based on readability but what you said makes sense here.